Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml

```diff
@@ -0,0 +1,719 @@
+import QtQuick 2.4
+import QtQuick.Window 2.2
+import QtQuick.Layouts 1.1
+import QtQuick.Controls 1.4
+import QtQuick.Controls.Styles 1.4
+import Esri.ArcGISRuntime 100.0
+import Esri.ArcGISExtras 1.1
+import "LeaderPosition.js" as Enums
+
+Item {
+    id: root
+    x: 0
+    y: 0
+
+    visible: false
+
+    // configurable properties
+    property bool autoAdjustWidth: true
+    property var leaderPosition: Enums.LeaderPosition.Bottom
+    property color borderColor: "#000000"
+    property int borderWidth: 2
+    property color backgroundColor: "#AFFFFFFF"
+    property color titleColor: "#000000"
+    property color detailColor: "#000000"
+    property int cornerRadius: 10
+    property int leaderHeight: 15
+    property int leaderWidth: 30
+    property int screenOffsetx: 0
+    property int screenOffsety: 0
+    property var calloutData
+
+    // internal properties
+    property int padding: 3
+    property real scaleFactor: System.displayScaleFactor
+    property real anchorPointx: 0
+    property real anchorPointy: 0
+    property point screenCoordinates
+    property var adjustedLeaderPosition: Enums.LeaderPosition.Bottom
+    property bool hidden: true
+    property real calloutMaxWidth: 210
+    property real calloutMaxHeight: 100
+    property real calloutMinWidth: calloutMaxWidth
+    property real calloutMinHeight: calloutMaxHeight
+    property real rectWidth: 0
+    property real rectHeight: 0
+    property real edgeBuffer: 10
+    property real calloutFramePadding: 2 * cornerRadius
+    property string platform: Qt.platform.os
+    property real halfRectWidth: rectWidth / 2
+    property real halfRectHeight: rectHeight / 2
+    property real halfLeaderWidth: leaderWidth / 2
+    property bool debug: false
+
+    Connections {
+        id: calloutConnection
+        target: calloutData
+        onScreenCoordinatesChanged: {
+            anchorPointx = calloutData.screenCoordinates.x;
+            anchorPointy = calloutData.screenCoordinates.y;
+
+//            if (Screen.devicePixelRatio > 1) {
+//                console.log("scaleFactor = ", scaleFactor);
+//                console.log("dividing by ", Screen.devicePixelRatio);
+//                anchorPointx = calloutData.screenCoordinates.x / Screen.devicePixelRatio;
+//                anchorPointy = calloutData.screenCoordinates.y / Screen.devicePixelRatio;
+//            }
+            // add any configured offsets
+            if (screenOffsetx !== 0)
+                anchorPointx += screenOffsetx
+            if (screenOffsety !== 0)
+                anchorPointy += screenOffsety
+
+            if (!hidden)
+                showCallout();
+        }
+
+        onImageUrlChanged: {
+            image.source = calloutData.imageUrl;
```

Comment:
I believe this can be automatically bound rather than manually doing it this way.